internal/core/software_renderer.rs

```diff
@@ -1082,18 +1116,35 @@ trait ProcessScene {
     fn process_gradient(&mut self, geometry: PhysicalRect, gradient: GradientCommand);
 }
 
-struct RenderToBuffer<'a, TargetPixel> {
-    buffer: &'a mut [TargetPixel],
-    stride: usize,
+struct RenderToBuffer<'a, TargetPixelBuffer> {
+    buffer: &'a mut TargetPixelBuffer,
     dirty_range_cache: Vec<core::ops::Range<i16>>,
     dirty_region: PhysicalRegion,
 }
 
-impl<T: TargetPixel> RenderToBuffer<'_, T> {
+impl<B: target_pixel_buffer::TargetPixelBuffer> RenderToBuffer<'_, B> {
     fn foreach_ranges(
         &mut self,
         geometry: &PhysicalRect,
-        mut f: impl FnMut(i16, &mut [T], i16, i16),
+        mut f: impl FnMut(i16, &mut [B::TargetPixel], i16, i16),
+    ) {
+        self.foreach_region(geometry, |buffer, rect, extra_left_clip, extra_right_clip| {
+            for l in rect.min_y()..rect.max_y() {
```

Comment:
I think this can be written
```suggestion
            for l in rect.y_range() {
```

Reply:
Nice - TIL :)